### ISSUE TITLE
Pod stuck in `ContainerCreating` and reporting `TLS handshake timeout` when deployed in fresh multi-node cluster

### ISSUE THREAD

**Issue by krakowski** [canonical/microk8s#3654]
#### Summary
After deploying a two node cluster, pods scheduled on the worker node are stuck in `ContainerCreating`.

```
 Events:                                                                                                                                                         
   Type     Reason                  Age                From               Message                                                                                
   ----     ------                  ----               ----               -------                                                                                
   Normal   Scheduled               102s               default-scheduler  Successfully assigned default/phpmyadmin-6c4dc6967d-k4v6v to kube-worker-1             
   Warning  FailedCreatePodSandBox  81s                kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network 
                                                                          for sandbox "9705456a848db7e2e12e323195b6eabda47fd5f144def34e156b362b79c236f0": plugin
                                                                          type="calico" failed (add): error getting ClusterInformation: Get "https://10.152.183.1:443/apis/crd.projectcalico.org/v1/clusterinformations/default": net/http: TLS handshake timeout                                                
   Normal   SandboxChanged          13s (x4 over 81s)  kubelet            Pod sandbox changed, it will be killed and re-created.
```

#### What Should Happen Instead?
Pods should start running and be ready on the worker node.

#### Reproduction Steps

I use the following Vagrantfile to create two virtual machines for the controller and worker node.

```ruby
Vagrant.configure("2") do |config|

  config.vm.box = "ubuntu/jammy64"

  config.vm.provider "virtualbox" do |vb|
    vb.memory = 4096
    vb.cpus = 2
  end
  
  config.vm.define "kube-controller-1" do |controller|
    controller.vm.hostname = "kube-controller-1"
    controller.vm.network "private_network", ip: "192.168.60.100"
    controller.vm.provider "virtualbox" do |vb|
      vb.name = "kube-controller-1"
    end
  end
  
  config.vm.define "kube-worker-1" do |worker|
    worker.vm.hostname = "kube-worker-1"
    worker.vm.network "private_network", ip: "192.168.60.200"
    worker.vm.provider "virtualbox" do |vb|
      vb.name = "kube-worker-1"
    end
  end
end
```

  1. Install `microk8s` in both machines using `snap`.
      ```
      sudo snap install microk8s --classic
      ```

  2. Enable `dns` and `metallb` addons on controller node.
      ```
      microk8s enable dns metallb:192.168.60.10-192.168.60.20
      ```

  3. Join worker node with controller.
      ```
      # Run on controller node
      microk8s add-node

      # Run on worker node
      microk8s join 192.168.60.100:25000/5f54b10eb07a459de66d40895918053d/2edc51feb413 --worker
      ```

  4. Enable Bitnami Helm repository on controller node.
      ```
      microk8s helm repo add bitnami https://charts.bitnami.com/bitnami
      ```

  5. Install phpmyadmin Helm chart.
      ```
      microk8s helm install phpmyadmin bitnami/phpmyadmin
      ```

#### Introspection Report

[inspection-report-20230105_232440.tar.gz](https://github.com/canonical/microk8s/files/10356318/inspection-report-20230105_232440.tar.gz)

#### Are you interested in contributing with a fix?
no

**Comment by berkayoz** [comment]
Hey @krakowski, thank you for opening this issue.

I've taken a look into your steps and could reproduce the issue using Vagrant. I believe the issue is related to the default network interface that is being set on the vagrant vms. The nodes use the `enp0s3 10.0.2.15/24` interface to advertise their addresses, which is the default interface of the system. This seems to be the root cause of this issue. Worker nodes try to proxy their requests to control-plane nodes, and can't go through the appropriate network. You might want to checkout this [page](https://microk8s.io/docs/configure-host-interfaces) to set a different network interface than the default one.

**Comment by krakowski** [comment]
Hey @berkayoz, sorry for the long delay.

After setting the private network ip inside all configuration files I can confirm that it works flawlessly. Here's what I changed within the config files in case somebody finds this issue and has the same problem. `192.168.60.100` is my controller's ip address which has to be changed accordingly.

`/var/snap/microk8s/current/args/kube-apiserver`
```
--advertise-address=192.168.60.100
```

`/var/snap/microk8s/current/args/kube-controller-manager`
```
--bind-address=192.168.60.100
```

`/var/snap/microk8s/current/args/kube-scheduler`
```
--bind-address=192.168.60.100
```

`/var/snap/microk8s/current/args/kubelet`
```
--node-ip=192.168.60.100
```

After this I restarted Kubernetes using `snap restart microk8s` and joined the worker node at which point everything worked.
